chapters/statements.tex

```diff
@@ -232,7 +232,8 @@ \subsubsection{Implicit Iteration Ranges}\label{implicit-iteration-ranges}
 \end{lstlisting}
 \end{example}
 
-The size of an array -- the iteration range -- is evaluated on entry to the \lstinline!for!-loop and the array size shall not change during the execution of the \lstinline!for!-loop.
+There may not be assignments to the entire array inside the loop, but there may be assignments to individual elements or ranges of elements.
+The size of an array -- the iteration range -- is evaluated on entry to the \lstinline!for!-loop and the array size will thus not change during the execution of the \lstinline!for!-loop.
```

Comment:
It would be better to have all the normative rules in one place.  Showing what I mean with a suggestion that is meant to be moved to the paragraph before the example:
```suggestion
… if the indices are of type \lstinline!Boolean!.
The iteration range is evaluated on entry to the \lstinline!for!-loop.
For an array to which the subscript is applied, there may not be assignments to the entire array inside the loop (but there may be assignments to individual elements or ranges of elements), ensuring that the iteration ranges determined upon entry will remain valid throughout the execution of the \lstinline!for!-loop.
If an iterator is used to subscript several…
```

Reply:
I understand the idea, but I have rewritten it instead of applying the suggestion.